source/dataman/BioXAS/BioXASDbUpgrade1Pt1.cpp

```diff
@@ -0,0 +1,209 @@
+#include "BioXASDbUpgrade1Pt1.h"
+#include "util/AMErrorMonitor.h"
+
+BioXASDbUpgrade1Pt1::BioXASDbUpgrade1Pt1(const QString &databaseNameToUpgrade, QObject *parent) :
+	AMDbUpgrade(databaseNameToUpgrade, parent)
+{
+
+}
+
+BioXASDbUpgrade1Pt1::~BioXASDbUpgrade1Pt1()
+{
+
+}
+
+QStringList BioXASDbUpgrade1Pt1::upgradeFromTags() const
+{
+	return QStringList();
+}
+
+bool BioXASDbUpgrade1Pt1::upgradeNecessary() const
+{
+	bool usingXRFDetector = databaseToUpgrade_->tableExists("BioXASScanConfigurationDbObject_table") && databaseToUpgrade_->columnExists("BioXASScanConfigurationDbObject_table", "usingXRFDetector");
+	bool sideConfiguration = !databaseToUpgrade_->tableExists("BioXASXASScanConfiguration_table") && databaseToUpgrade_->tableExists("BioXASSideXASScanConfiguration_table");
+	bool mainConfiguration = !databaseToUpgrade_->tableExists("BioXASXASScanConfiguration_table") && databaseToUpgrade_->tableExists("BioXASMainXASScanConfiguration_table");
+
+	bool upgradeNeeded = (usingXRFDetector || sideConfiguration || mainConfiguration);
+	return upgradeNeeded;
+}
+
+bool BioXASDbUpgrade1Pt1::upgradeImplementation()
+{
+	// Begin upgrade.
+
+	databaseToUpgrade_->startTransaction();
+
+	// Remove usingXRFDetector column from BioXASScanConfigurationDbObject_table, if it exists. This column is no longer used.
+
+	if (databaseToUpgrade_->tableExists("BioXASScanConfigurationDbObject_table")) {
+		if (databaseToUpgrade_->columnExists("BioXASXASScanConfigurationDbObject_table", "usingXRFDetector")) {
+			if (!AMDbUpgradeSupport::removeColumn(databaseToUpgrade_, "BioXASScanConfigurationDbObject_table", "usingXRFDetector")) {
+				AMErrorMon::alert(this, BIOXASDBUPGRADE1PT1_COULD_NOT_REMOVE_COLUMN, "Could not remove the usingXRFDetector column from BioXASScanConfigurationDbObject_table.");
+				databaseToUpgrade_->rollbackTransaction();
+				return false;
+			}
+		}
+	}
+
+	// Create and populate BioXASXASScanConfiguration_table, if it doesn't already exist.
+
+	if (!databaseToUpgrade_->tableExists("BioXASXASScanConfiguration_table")) {
+		QStringList names = QStringList() << "AMDbObjectType" << "thumbnailCount" << "thumbnailFirstId" << "name" << "detectorConfigurations" << "axisControlInfos" << "configurationDbObject" << "header";
+		QStringList types = QStringList() << "TEXT" << "INTEGER" << "INTEGER" << "TEXT" << "TEXT" << "TEXT" << "TEXT" << "TEXT";
+
+		if (!databaseToUpgrade_->ensureTable("BioXASXASScanConfiguration", names, types)) {
```

Comment:
I would use the `AMDbUpgradeSupport::addTable()` method instead since it handles all the support table changes as well.